### PR TITLE
mgr: Configure Py root logger for Mgr modules

### PR DESCRIPTION
**Pull request by votdev**
Add the CPlusPlusHandler to the Python root logger, thus the Mgr module itself and all 3rd party libraries will log to the Ceph log. This may help to identify problems in used 3rd party libraries.

Signed-off-by: Volker Theile <vtheile@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/mgr_module.py [hunk 1]
```diff
@@ -59,22 +59,41 @@ def emit(self, record):
...
+    :type module_inst: instance
+    :param module_name: The module name.
+    :type module_name: str
+    :return: Return the logger with the specified name.
+    """
+    logger = logging.getLogger(module_name)
+    # Don't filter any logs at the python level, leave it to C++.
+    logger.setLevel(logging.NOTSET)
 
     # FIXME: we should learn the log level from C++ land, and then
```
Comment (sebastian-philipp) [inline src/pybind/mgr/mgr_module.py]: afaik this FIXME belongs to `logger.setLevel` above

File: src/pybind/mgr/mgr_module.py [hunk 2]
```diff
@@ -59,22 +59,41 @@ def emit(self, record):
...
     # avoid calling the C++ level log when we know a message is of
     # an insufficient level to be ultimately output
-    logger.addHandler(CPlusPlusHandler(module_inst))
+
+    # Add handler to the root logger, thus this module and all
+    # 3rd party libraries will log their messages to the Ceph log.
+    root_logger = logging.getLogger()
```
Comment (sebastian-philipp) [inline src/pybind/mgr/mgr_module.py]: Does it make sense to set the level of `root_logger` to ERROR?

**Comment by sebastian-philipp** [comment]
I'd be :+1:  for backporting this to nautilus.

File: src/pybind/mgr/mgr_module.py (hunk 2 above)
Comment (votdev) [reply src/pybind/mgr/mgr_module.py]: Done, but keep in mind how the effective log level is determined. If a 3rd party library set a other level than ``ogging.NOTSET``, then this level will be used and no further filtering in ancestor loggers is done.

File: src/pybind/mgr/mgr_module.py (hunk 1 above)
Comment (votdev) [reply src/pybind/mgr/mgr_module.py]: Fixed

File: src/pybind/mgr/mgr_module.py (hunk 2 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/mgr_module.py]: was meant to just be a default for 3rd party libraries. for example, the `requests` library is logging too much on the INFO level.

**Comment by sebastian-philipp** [comment]
make check failed: http://tracker.ceph.com/issues/38892

**Comment by badone** [comment]
jenkins test make check

**Pull request merged**